packages/plugin-core/src/commands/CopyAsCommand.ts

```diff
@@ -0,0 +1,85 @@
+import { assertUnreachable } from "@dendronhq/common-all";
+import {
+  CopyAsFormat,
+  getAllCopyAsFormat,
+  JSONV2PodConfig,
+  MarkdownV2PodConfig,
+  PodExportScope,
+  PodV2Types,
+} from "@dendronhq/pods-core";
+import _ from "lodash";
+import { PodCommandFactory } from "../components/pods/PodCommandFactory";
+import { PodUIControls } from "../components/pods/PodControls";
+import { DENDRON_COMMANDS } from "../constants";
+import { VSCodeUtils } from "../vsCodeUtils";
+import { BaseCommand, CodeCommandInstance } from "./base";
+
+type CommandOutput = void;
+type CommandInput = CodeCommandInstance;
+type CommandOpts = CodeCommandInstance;
+
+/**
+ * Command that will find the appropriate export command to run, and then run
+ * it. This is the UI entry point for all export pod functionality.
+ */
+export class CopyAsCommand extends BaseCommand<
+  CommandOpts,
+  CommandOutput,
+  CommandInput
+> {
+  public format: CopyAsFormat[];
+  key = DENDRON_COMMANDS.COPY_AS.key;
+
+  constructor(_name?: string) {
+    super(_name);
+    this.format = getAllCopyAsFormat();
+  }
+
+  async sanityCheck(_opts?: Partial<CodeCommandInstance> | undefined) {
+    if (_.isUndefined(VSCodeUtils.getActiveTextEditor())) {
+      return "you must have a note open to execute this command";
+    }
+    return;
+  }
+
+  async gatherInputs() {
```

Comment:
low hanging enhancement idea for later:
allow command opts to specify the format so we can define keybindings for each `CopyAsFormat`. (could be tied to a context menu this way too)

Reply:
https://github.com/dendronhq/dendron/pull/3544/commits/0ec47f1228fcfa90c129054a6e1232fa86030e5c